File: app/src/main/java/com/snowdango/numac/NumAcMain/ButtonClickEvent.java
package com.snowdango.numac.NumAcMain;

import android.content.Intent;
import android.os.Handler;
import android.util.Log;
import android.widget.TextView;

import static com.snowdango.numac.NumAcMain.NumAcActivity.dataBaseHelper;
import static com.snowdango.numac.NumAcMain.NumAcFragment.textPut;

public class ButtonClickEvent {

    private Handler handler = new Handler();

    public Intent onClickEvents(final TextView textView) {
        Intent intent = null;

        Runnable updateText = new Runnable() {
            @Override
            public void run() {
                textView.setText("");
                textPut = true;
            }
        };

        String command = (String) textView.getText();

        try {
            textView.setText("Load");
            textPut = false;
            String[] info = dataBaseHelper.getPackageAndClass(dataBaseHelper, command);

            if (info[0] != null) {
                intent = new Intent();
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                intent.setClassName(info[0], info[1]);
            }

        } catch (Exception e) {
            Log.d("command", command);
            textView.setText("Error");
            textPut = false;
            intent = null;
            handler = new Handler();
            handler.postDelayed(updateText, 1000);
        } finally {
            return intent;
        }
    }
}

File: app/src/main/java/com/snowdango/numac/NumAcMain/ButtonClickEvent.kt
package com.snowdango.numac.NumAcMain

import android.content.Intent
import android.os.Handler
import android.util.Log
import android.widget.TextView

class ButtonClickEvent {
    private var handler = Handler()
    fun onClickEvents(textView: TextView): Intent? {
        var intent: Intent? = null
        val updateText = Runnable {
            textView.text = ""
            NumAcFragment.textPut = true
        }
        val command = textView.text as String
        try {
            textView.text = "Load"
            NumAcFragment.textPut = false
            val info: Array<String> = NumAcActivity.dataBaseHelper!!.getPackageAndClass(NumAcActivity.dataBaseHelper!!, command)
            intent = Intent()
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
            intent.setClassName(info[0]!!, info[1]!!)
        } catch (e: Exception) {
            Log.d("command", command)
            textView.text = "Error"
            NumAcFragment.textPut = false
            intent = null
            handler = Handler()
            handler.postDelayed(updateText, 1000)
        } finally {
            return intent
        }
    }
}